cpp/tests/c_api/random_walks_test.c

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "c_test_utils.h" /* RUN_TEST */
+
+#include <cugraph_c/cugraph_api.h>
+
+#define NUM_PATHS 2
+#define MAX_DEPTH 4
+
+/* sample graph:
+    0 --(.1)--> 1 --(1.1)--> 4
+   /|\       /\ |            |
+    |       /   |            |
+   (5.1) (3.1)(2.1)        (3.2)
+    |   /       |            |
+    | /        \|/          \|/
+    2 --(4.1)-->3 --(7.2)--> 5
+*/
+
+/* positive test RW call flow*/
+int test_random_walks_1()
+{
+  typedef int32_t vertex_t;
+  typedef int32_t edge_t;
+  typedef float weight_t;
+
+  cugraph_error_t ret_code = CUGRAPH_SUCCESS;
+  size_t num_edges         = 8;
+  size_t num_vertices      = 6;
+
+  vertex_t h_src[] = {0, 1, 1, 2, 2, 2, 3, 4};
+  vertex_t h_dst[] = {1, 3, 4, 0, 1, 3, 5, 5};
+  weight_t h_wgt[] = {0.1f, 2.1f, 1.1f, 5.1f, 3.1f, 4.1f, 7.2f, 3.2f};
+
+  cugraph_raft_handle_t* p_handle = NULL;
+  cugraph_device_buffer_t dbuf_src;
+  cugraph_device_buffer_t dbuf_dst;
+  cugraph_device_buffer_t dbuf_wgt;
+  cugraph_graph_envelope_t* p_graph_envelope = NULL;
+
+  data_type_id_t vertex_tid = INT32;
+  data_type_id_t edge_tid   = INT32;
+  data_type_id_t weight_tid = FLOAT32;
+
+  /* RW args:*/
+  cugraph_rw_ret_t rw_ret;
+  cugraph_device_buffer_t dbuf_start;
+  size_t num_paths                          = 2;
+  size_t max_depth                          = 4;
+  bool_t flag_use_padding                   = FALSE;
+  cugraph_unique_ptr_t* p_sampling_strategy = NULL;
+  vertex_t h_start[]                        = {0, 2};
+
+  p_handle = cugraph_create_handle();
+  runtime_assert(p_handle != NULL, "raft handle creation failed.");
```

Comment:
since this isn't a real assert (the implementation I saw above seems to just print to stderr), the final return value of the test needs to reflect if this fails, but it's just being ignored.  Same for all other `runtime_assert` calls that aren't setting a flag of some sort.

Reply:
We can modify this in the next release.